generate/efi/Makefile.config

```diff
@@ -111,7 +111,8 @@ LDFLAGS	= \
 	-Bsymbolic
 ifeq ($(strip $(SHARED)), true)
 LDFLAGS	+= \
-	-shared
+	-shared\
+	--no-undefined
 endif
```

Comment:
I like this change.
However we need to remove SHARED environment and fix libgcc.a order in the meanwhile.
I have this fixed and will include this in the pull request:
https://github.com/acpica/acpica/pull/149